indexing-service/src/main/java/io/druid/indexing/overlord/autoscaling/PendingTaskBasedResourceManagementStrategy.java

```diff
@@ -0,0 +1,360 @@
+        log.warn("No workerConfig available, cannot provision new workers.");
+        return false;
+      }
+
+      final List<String> workerNodeIds = workerConfig.getAutoScaler().ipToIdLookup(
+          Lists.newArrayList(
+              Iterables.<ZkWorker, String>transform(
+                  zkWorkers,
+                  new Function<ZkWorker, String>()
+                  {
+                    @Override
+                    public String apply(ZkWorker input)
+                    {
+                      return input.getWorker().getIp();
+                    }
+                  }
+              )
+          )
+      );
+      currentlyProvisioning.removeAll(workerNodeIds);
+      if (currentlyProvisioning.isEmpty()) {
+        int want = getScaleUpNodeCount(
+            runner.getConfig(),
+            workerConfig,
+            pendingTasks,
+            zkWorkers
+        );
+        while (want > 0) {
+          final AutoScalingData provisioned = workerConfig.getAutoScaler().provision();
+          final List<String> newNodes;
+          if (provisioned == null || (newNodes = provisioned.getNodeIds()).isEmpty()) {
+            break;
+          } else {
+            currentlyProvisioning.addAll(newNodes);
+            lastProvisionTime = new DateTime();
+            scalingStats.addProvisionEvent(provisioned);
+            want -= provisioned.getNodeIds().size();
+            didProvision = true;
+          }
+        }
+      } else {
+        Duration durSinceLastProvision = new Duration(lastProvisionTime, new DateTime());
+        log.info("%s provisioning. Current wait time: %s", currentlyProvisioning, durSinceLastProvision);
+        if (durSinceLastProvision.isLongerThan(config.getMaxScalingDuration().toStandardDuration())) {
+          log.makeAlert("Worker node provisioning taking too long!")
+             .addData("millisSinceLastProvision", durSinceLastProvision.getMillis())
+             .addData("provisioningCount", currentlyProvisioning.size())
+             .emit();
+
+          workerConfig.getAutoScaler().terminateWithIds(Lists.newArrayList(currentlyProvisioning));
+          currentlyProvisioning.clear();
+        }
+      }
+
+      return didProvision;
+    }
+  }
+
+  int getScaleUpNodeCount(
+      final RemoteTaskRunnerConfig remoteTaskRunnerConfig,
+      final WorkerBehaviorConfig workerConfig,
+      final Collection<Task> pendingTasks,
+      final Collection<ZkWorker> zkWorkers
+  )
+  {
+    final int minWorkerCount = workerConfig.getAutoScaler().getMinNumWorkers();
+    final int maxWorkerCount = workerConfig.getAutoScaler().getMaxNumWorkers();
+    final Predicate<ZkWorker> isValidWorker = ResourceManagementUtil.createValidWorkerPredicate(config);
+    final int currValidWorkers = Collections2.filter(zkWorkers, isValidWorker).size();
+    int moreWorkersNeeded = currValidWorkers == 0 ? minWorkerCount : getWorkersNeededToAssignTasks(
```

Comment:
essentially what I'm asking is: if the number of workers needed for the tasks is larger than the minWorkerCount, can you change this logic so that the number needed to launch the task is favored?